### PR TITLE
Frozen headers

### PR DESCRIPTION
**Pull request by Blackbaud-AlexKingman**
Addresses #84 

Note:
- Using the `position: sticky` property. Not supported by IE11.
- Table header borders do not "stick", so we had to use `box-shadow` to emulate the borders. This shifted all the rows over by a pixel, so we had to blow away the visual screenshots to reset them to the new layout.

### CODE REVIEW

File: src/app/visual/grid/grid-visual.component.html [hunk 1]
```diff
@@ -140,3 +140,28 @@ <h1>Grid with inline help</h1>
     This is a string column. Click <a href="#">here</a> to learn more.
   </sky-popover>
 </div>
+
+<h1>Grid inside scrollable container</h1>
+
+<div
+  id="screenshot-grid-scrollable"
```
Comment (Blackbaud-AlexKingman) [inline src/app/visual/grid/grid-visual.component.html]: This new section was added purely for the visual tests.

File: src/app/public/modules/grid/grid.component.scss [hunk 2]
```diff
@@ -46,10 +46,14 @@ $sky-grid-resize-handle-width: 5px;
...
   position: sticky;
   top:0;
-  box-shadow: 0 0 0 1px #cdcfd2;
+  box-shadow: 1px 0 0 1px #cdcfd2;
   border-right-width: 0px;
   padding: 0;
 
+  // After adding the box-shadow false borders above, the lack of borders results in a 1px gap.
+  // This style creates an invisible border to prevent any visual tests from failing.
+  border-left: 1px solid $sky-color-white;
```
Comment (Blackbaud-TrevorBurch) [inline src/app/public/modules/grid/grid.component.scss]: We already have this for the first child but we use "transparent". Any reason we just don't do transparent here and remove the first child one?

Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/grid/grid.component.scss]: Good call. I've removed the :first-child declaration.

File: src/app/public/modules/grid/grid.component.scss [hunk 3]
```diff
@@ -48,22 +44,21 @@ $sky-grid-resize-handle-width: 5px;
 }
 
 .sky-grid-heading {
-  position: relative;
-  @include sky-border(dark, top, bottom, left, right);
-  border-right-width: 0px;
+  position: sticky;
+  top:0;
+  box-shadow: 0 0 0 1px #cdcfd2;
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/grid/grid.component.scss]: This color value is a SCSS variable: https://developer.blackbaud.com/skyux/design/styles/color#text

Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/grid/grid.component.scss]: Also, could you add a comment explaining the use of box-shadow (with a link to the StackOverflow)?

**Comment by Blackbaud-BobbyEarl** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-grids?_sp.spa=pr-frozen-headers](https://host.nxt.blackbaud.com/skyux-grids?_sp.spa=pr-frozen-headers)

**Comment by Blackbaud-AlexKingman** [comment]
Closing in favor ot #89

**Pull request closed without merging**